asmcomp/arm64/emit.mlp

```diff
@@ -1021,16 +1068,57 @@ let fundecl fundecl =
   let num_call_gc, num_check_bound =
     num_call_gc_and_check_bound_points env
   in
+
+  (* Dynamic stack checking *)
+  let stack_threshold_size = Config.stack_threshold * 8 in (* bytes *)
+  let { max_stack_size; contains_nontail_calls } =
+    preproc_fun env fundecl.fun_body fundecl.fun_name
+  in
+  let stack_check_size = ref 0 in
+  let handle_overflow = ref None in
+  if contains_nontail_calls || max_stack_size >= stack_threshold_size then begin
+    let overflow = new_label () and ret = new_label () in
+    let threshold_offset = Domainstate.stack_ctx_words * 8 + stack_threshold_size in
```

Comment:
We could maintain the stack threshold (a pointer into the current stack) in the domain state. We can load this address from the domain state, and should shave off one instruction. This should obviate the need for using `Config.stack_threshold` and `Domainstate.stack_ctx_words` in `emit.mlp`. These variables contain information that is maintained by the runtime, and it will be better if they don't appear in the codegen.

Reply:
@ctk21 pointed to me that this proposal cannot do away with the `add` instruction due to the fact that `max_stack_size` is per function. 

Either way, these optimisations are may be explored in a separate PR. I'm resolving this conversation.